lua/orgmode/org/syntax.lua

```diff
@@ -36,7 +42,77 @@ local function add_todo_keywords_to_spellgood()
   end
 end
 
+local function _display_cookie_virt_text(cookie, checked, total)
+  local row, column = cookie:start()
+  local cookie_text = vim.treesitter.get_node_text(cookie, 0)
+
+  -- TODO: fix this in treesitter
+  local _, leading_spaces = cookie_text:find('^%s+')
+  cookie_text = vim.trim(cookie_text)
+
+  local text
+  if cookie_text == '[/]' then
+    text = string.format('[%d/%d]', checked, total)
+  else
+    if total > 0 then
+      text = string.format('[%d%%]', (100 * checked) / total)
+    else
+      text = '[0%]'
+    end
+  end
+  vim.api.nvim_buf_set_extmark(0, cookie_namespace, row, column + (leading_spaces or 0), {
```

Comment:
Emacs orgmode is actually modifying the content for this instead of using something like virtual text. That's why they require `[/]` to be added to the end. I'd love to propose 2 solutions to this:

1. Do the same thing that emacs does (Really replace the text)
2. Implement this as a virtual text without a need for a user to type `[/]` 

Current solution is kind of a hybrid, and I think having empty `[/]` as a content will cause confusion.
I'd love to go with 1st solution, since it copies the emacs orgmode functionality which I'm trying to achieve most of the time.
I know that 2nd solution is simpler, and we could have it enabled by default, and to allow it to be disabled through config.
If first one is too complicated, we can go with 2nd one.

Reply:
:+1: I didn't know emacs modifies the file.
shouldn't be that hard to do.

Reply:
Updating the text in the buffer is very expensive.. it gets very  slow even with only a couple cookies.
Maybe we use virtual text and only update the actual text on save?

Reply:
@lukas-reineke how are you updating it? It should update only when you check/uncheck something, maybe you are triggering it more often. Maybe if you have file with several thousand lines, but even than shouldn't cause that many issues.

Reply:
I was updating all of them before, but yeah without virt text it makes much more sense to only trigger this on checkbox toggle.
I rewrote the logic to do this.
Added bonus, checkboxes update their parents now as well.